### PR TITLE
[checkbox-ce-oem] Add SPI Test (New)

### PR DESCRIPTION
**Pull request by LiaoU3**
## Description
Add SPI Test. Originally from [checkbox-provider-baoshan](https://git.launchpad.net/~baoshan-team/baoshan/+git/checkbox-provider-baoshan/tree/units/peripheral/jobs.pxu#n46)

## Resolved issues
N/A

## Documentation
N/A

## Tests
https://certification.canonical.com/hardware/202405-34024/submission/375779/

### CODE REVIEW

File: contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/spidev_test.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        SystemExit: If no output is reported or
+        if the content is not consistent.
+    """
+
+    # Construct the command to run the SPI test
+    cmd = "spidev_test -D {} -p 'Canonical SPI Test' -v".format(spi_path)
+    print(f"Run command: {cmd}\n")
+
+    # Run the command and capture the output
+    spi_ret = runcmd([cmd])
```
Comment (stanley31huang) [inline contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/spidev_test.py:47]: I think we should split the cmd here by shlex

Comment (LiaoU3) [reply contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/spidev_test.py]: Fiexd

File: contrib/checkbox-ce-oem/checkbox-provider-ce-oem/units/spi/jobs.pxu [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+environ:
+    SPI_DEVICE_COUNT
+command:
+    spidev_test.py -d "${SPI_DEVICE_COUNT:-1}"
+
+unit: template
+template-resource: ce-oem-spi-list
+template-unit: job
+template-id: ce-oem-spi/spi-loopback-device
+_template-_summary: Test the SPI bus functionality on devices
```
Comment (weizhenwu) [inline contrib/checkbox-ce-oem/checkbox-provider-ce-oem/units/spi/jobs.pxu:23]:
```suggestion
_template-summary: Test the SPI bus functionality on devices
```

**Review by weizhenwu — CHANGES_REQUESTED** [review]

File: contrib/checkbox-ce-oem/checkbox-provider-ce-oem/units/spi/jobs.pxu (hunk 2 above)
Comment (LiaoU3) [reply contrib/checkbox-ce-oem/checkbox-provider-ce-oem/units/spi/jobs.pxu]: Commited. It's weird that `manage.py validate` did not find this error.

File: contrib/checkbox-ce-oem/checkbox-provider-ce-oem/units/spi/jobs.pxu [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+requires:
+    manifest.has_spi == 'True'
+    manifest.has_spi_loopback == "True"
+_summary: Test the SPI bus functionality on {device}
+_purpose:
+    Check if the SPI devices can be used via spidev_test tool to loopback test
+_description:
+    This test will send a test_string with spidev_test tool to check whether the
+    test_string will be received via the loopback function.
+    Please connect the SPI0_MO & SPI0_MI  pins with jumper or wiret.
```
Comment (LiaoU3) [inline contrib/checkbox-ce-oem/checkbox-provider-ce-oem/units/spi/jobs.pxu:41]:
```suggestion
    Please connect the MISO & MOSI  pins with jumper or wire.
```

**Review by LiaoU3 — COMMENTED** [review]
Please see my comment.

File: contrib/checkbox-ce-oem/checkbox-provider-ce-oem/units/spi/jobs.pxu (hunk 3 above)
Comment (LiaoU3) [reply contrib/checkbox-ce-oem/checkbox-provider-ce-oem/units/spi/jobs.pxu]: Thank you so much!

File: contrib/checkbox-ce-oem/checkbox-provider-ce-oem/units/spi/testplan.pxu [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+# This testplan should be removed when we merge before and after suspend testplan
+id: after-suspend-ce-oem-spi-automated
+unit: test plan
+_name: After Suspend Serial Peripheral Interface (SPI) Automated Tests
+_description:
+    After Suspend Serial Peripheral Interface (SPI) Automated Tests
+bootstrap_include:
+    ce-oem-spi-list
+include:
+    ce-oem-spi/detect
```
Comment (stanley31huang) [inline contrib/checkbox-ce-oem/checkbox-provider-ce-oem/units/spi/testplan.pxu:52]:
```suggestion
    after-suspend--ce-oem-spi/detect
```

**Review by stanley31huang — APPROVED** [review]
LGTM

File: contrib/checkbox-ce-oem/checkbox-provider-ce-oem/units/spi/testplan.pxu (hunk 4 above)
Comment (rickwu666666) [reply contrib/checkbox-ce-oem/checkbox-provider-ce-oem/units/spi/testplan.pxu]: This modification seems not correct.

**Review by weizhenwu — DISMISSED** [review]
+1

File: contrib/checkbox-ce-oem/checkbox-provider-ce-oem/units/spi/testplan.pxu [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+# This testplan should be removed when we merge before and after suspend testplan
+id: after-suspend-ce-oem-spi-automated
+unit: test plan
+_name: After Suspend Serial Peripheral Interface (SPI) Automated Tests
+_description:
+    After Suspend Serial Peripheral Interface (SPI) Automated Tests
+bootstrap_include:
+    ce-oem-spi-list
+include:
+    after-suspend--ce-oem-spi/detect
```
Comment (LiaoU3) [inline contrib/checkbox-ce-oem/checkbox-provider-ce-oem/units/spi/testplan.pxu:52]:
```suggestion
    after-suspend-ce-oem-spi/detect
```

Comment (rickwu666666) [reply contrib/checkbox-ce-oem/checkbox-provider-ce-oem/units/spi/testplan.pxu]: LGTM

**Review by rickwu666666 — APPROVED** [review]
LGTM

**Pull request merged**